### PR TITLE
Build all plugins as independent jars.

### PR DESCRIPTION
**Pull request by asuresh8**
### Description
Build all plugins an independent jars
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: build-resources.gradle [hunk 1]
```diff
@@ -6,8 +6,71 @@
...
         project(':data-prepper-api'),
         project(':data-prepper-core'),
-        project('data-prepper-plugins:common'),
-        project('data-prepper-expression'),
+        project(':data-prepper-plugins:common'),
+        project(':data-prepper-expression'),
         project(':data-prepper-logstash-configuration')
 ]
-ext.mavenArtifactProjects = [project(':data-prepper-api')]
+ext.mavenArtifactProjects = [
```
Comment (dlvenable) [inline build-resources.gradle:13]: Can we just publish all jars and thus remove this entirely?

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: build-resources.gradle (hunk 1 above)
Comment (asuresh8) [reply build-resources.gradle]: Yes, I have updated the PR by changing the core projects to reflect all the top level projects we want to publish. Then I've added a release task that recursively goes through the core projects and publish them.

File: build.gradle [hunk 2]
```diff
@@ -20,8 +20,10 @@ allprojects {
     apply plugin: 'checkstyle'
     apply plugin: 'com.diffplug.spotless'
     apply plugin: 'com.github.jk1.dependency-license-report'
+    apply plugin: 'maven-publish'
 
     group = 'org.opensearch.dataprepper'
+    version = '2.7.0-SNAPSHOT'
```
Comment (dlvenable) [inline build.gradle:26]: This value already comes from the `gradle.properties`. Let's use only that one as the source of truth.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (asuresh8) [reply build.gradle]: Done. I'm still learning what Gradle plumbs in and doesn't plumb in.

**Review by dlvenable — APPROVED** [review]
Thanks!

**Review by oeyh — APPROVED** [review]

**Pull request merged**